distributed/comm/addressing.py

```diff
@@ -301,3 +301,13 @@ def address_from_user_args(  # type: ignore[no-untyped-def]
         addr = protocol + "://" + addr.split("://")[-1]
 
     return addr
+
+
+def is_loopback_address(addr: str) -> bool:
+    """Return True if the given ipv4 or ipv6 address is a loopback address (e.g. the
+    server is on localhost); False otherwise
+    """
+    return addr.startswith("127.0.0.") or addr in (
+        "::1",
+        "0000:0000:0000:0000:0000:0000:0000:0001",
+    )
```

Comment:
[ipaddress](https://docs.python.org/3/library/ipaddress.html)
```python
import ipaddress
assert ipaddress.ip_interface('127.0.0.1').is_loopback
```
Appears to cover the ipv6 checks here as well.